waku/v2/protocol/peer_exchange/waku_peer_exchange_test.go

```diff
@@ -185,3 +213,196 @@ func TestRetrieveFilteredPeerExchangePeers(t *testing.T) {
+	require.NoError(t, err)
+	d1.SetHost(host1)
+
+	// Mount peer exchange
+	pxPeerConn1 := discv5.NewTestPeerDiscoverer()
+	px1, err := NewWakuPeerExchange(d1, pxPeerConn1, nil, prometheus.DefaultRegisterer, utils.Logger())
+	require.NoError(t, err)
+	px1.SetHost(host1)
+
+	// Test WithPeerAddr()
+	params := new(PeerExchangeRequestParameters)
+	params.host = px1.h
+	params.log = px1.log
+	params.pm = px1.pm
+
+	optList := DefaultOptions(px1.h)
+	optList = append(optList, WithPeerAddr(host1.Addrs()[0]))
+	for _, opt := range optList {
+		err := opt(params)
+		require.NoError(t, err)
+	}
+
+	require.Equal(t, host1.Addrs()[0], params.peerAddr)
+
+	// Test WithFastestPeerSelection()
+	optList = DefaultOptions(px1.h)
+	optList = append(optList, WithFastestPeerSelection(host1.ID()))
+	for _, opt := range optList {
+		err := opt(params)
+		require.NoError(t, err)
+	}
+
+	require.Equal(t, peermanager.LowestRTT, params.peerSelectionType)
+	require.Equal(t, host1.ID(), params.preferredPeers[0])
+
+}
+
+func TestRetrieveProvidePeerExchangeWithPMAndPeerAddr(t *testing.T) {
+	log := utils.Logger()
+
+	// H1 + H2 with discovery on
+	host1, d1 := createHostWithDiscv5(t)
+	host2, d2 := createHostWithDiscv5(t, d1.Node())
+
+	// H3
+	ps3, err := pstoremem.NewPeerstore()
+	require.NoError(t, err)
+	wakuPeerStore3 := wps.NewWakuPeerstore(ps3)
+	host3, _, _ := tests.CreateHost(t, libp2p.Peerstore(wakuPeerStore3))
+
+	defer d1.Stop()
+	defer d2.Stop()
+	defer host1.Close()
+	defer host2.Close()
+	defer host3.Close()
+
+	err = d1.Start(context.Background())
+	require.NoError(t, err)
+
+	err = d2.Start(context.Background())
+	require.NoError(t, err)
+
+	// Prepare peer manager for host3
+	pm3 := peermanager.NewPeerManager(10, 20, log)
+	pm3.SetHost(host3)
+	pxPeerConn3, err := peermanager.NewPeerConnectionStrategy(pm3, 30*time.Second, utils.Logger())
+	require.NoError(t, err)
+	pxPeerConn3.SetHost(host3)
+	err = pxPeerConn3.Start(context.Background())
+	require.NoError(t, err)
+
+	// mount peer exchange
+	pxPeerConn1 := discv5.NewTestPeerDiscoverer()
+	px1, err := NewWakuPeerExchange(d1, pxPeerConn1, nil, prometheus.DefaultRegisterer, utils.Logger())
+	require.NoError(t, err)
+	px1.SetHost(host1)
+
+	//pxPeerConn3 := discv5.NewTestPeerDiscoverer()
+	px3, err := NewWakuPeerExchange(nil, pxPeerConn3, pm3, prometheus.DefaultRegisterer, utils.Logger())
+	require.NoError(t, err)
+	px3.SetHost(host3)
+
+	err = px1.Start(context.Background())
+	require.NoError(t, err)
+
+	time.Sleep(5 * time.Second)
+
+	// Check peer store of the client doesn't contain peer2 before the request
+	require.False(t, slices.Contains(host3.Peerstore().Peers(), host2.ID()))
+
+	// Construct multi address like example "/ip4/0.0.0.0/tcp/30304/p2p/16Uiu2HAmBu5zRFzBGAzzMAuGWhaxN2EwcbW7CzibELQELzisf192"
+	host1MultiAddr, err := multiaddr.NewMultiaddr(host1.Addrs()[0].String() + "/p2p/" + host1.ID().String())
```

Comment:
more of a nitpick but this can be done via Encapsulate()
```suggestion
	hostInfo, err := multiaddr.NewMultiaddr(fmt.Sprintf("/p2p/%s", host1.Host().ID()))
	require.NoError(t, err)
	host1MultiAddr, err := host1.Addrs()[0].Encapsulate(hostInfo)
```

Reply:
Fixed at https://github.com/waku-org/go-waku/pull/1046/commits/2e8a07a252261aafdba60c33c15000056a03cc6d